script.embuary.helper/resources/lib/cinema_mode.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python
+# coding: utf-8
+
+#################################################################################################
+
+import xbmc
+import xbmcgui
+import xbmcvfs
+import random
+import os
+
+from resources.lib.helper import *
+
+#################################################################################################
+
+class CinemaMode(object):
+    def __init__(self,dbid,dbtype):
+        self.trailer_count = xbmc.getInfoLabel('Skin.String(TrailerCount)') if xbmc.getInfoLabel('Skin.String(TrailerCount)') != '0' else False
+        self.intro_path = xbmc.getInfoLabel('Skin.String(IntroPath)')
+
+        self.dbid = dbid
+        self.dbtype = dbtype
+
+        if not self.dbid or not self.dbtype:
+            for i in range(30):
```

Comment:
Because I've learned from the Emby integration that sometimes stuff doesn't get returned fast enough. On slower devices and on skins with longer WindowClose animations.